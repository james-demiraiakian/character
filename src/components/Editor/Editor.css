.inputs {
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  color: white;
  text-shadow: 0 0 1px black, 0 0 1px black, 0 0 1px black, 0 0 1px black, 0 0 1px black,
    0 0 1px black, 0 0 1px black, 0 0 1px black, 0 0 1px black;
  display: flex;
  flex-direction: column;
  height: 19vw;
  width: 40vw;
  align-items: center;
  margin-top: 10vw;
}

.form {
  margin: 10px;
  align-items: center;
}

.counter {
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  color: white;
  text-shadow: 0 0 1px black, 0 0 1px black, 0 0 1px black, 0 0 1px black, 0 0 1px black,
    0 0 1px black, 0 0 1px black, 0 0 1px black, 0 0 1px black;
  display: flex;
  flex-direction: column;
  width: 41vw;
  align-items: center;
  margin-top: 20px;
  margin-bottom: 20px;
}
