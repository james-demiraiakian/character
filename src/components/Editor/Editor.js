import React from 'react';

import './Editor.css';

export default function Editor({
  head,
  setHead,
  body,
  setBody,
  legs,
  setLegs,
  phrase,
  setPhrase,
  setPhraseList,
}) {
  const phrases = { phrase };
  const handlePhrase = () => {
    setPhraseList((prevState) => [...prevState, phrase]);
    setPhrase('');
  };

  return (
    <div className="editor">
      <div className="form">
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          <option value="head-1">Head 1</option>
          <option value="head-2">Head 2</option>
          <option value="head-3">Head 3</option>
        </select>
      </div>
      <div className="form">
        <select value={body} onChange={(e) => setBody(e.target.value)}>
          <option value="body-1">Body 1</option>
          <option value="body-2">Body 2</option>
          <option value="body-3">Body 3</option>
        </select>
      </div>
      <div className="form">
        <select value={legs} onChange={(e) => setLegs(e.target.value)}>
          <option value="legs-1">Legs 1</option>
          <option value="legs-2">Legs 2</option>
          <option value="legs-3">Legs 3</option>
        </select>
      </div>
      <div className="form">
        <input type="text" value={phrase} onChange={(e) => setPhrase(e.target.value)} />
        <label>Phrase</label>
        <button onClick={handlePhrase}>Save</button>
      </div>
    </div>
  );
}
