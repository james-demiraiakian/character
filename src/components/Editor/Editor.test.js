import { render, screen } from '@testing-library/react';
import Editor from './Editor';

test('test of dom render', () => {
  const container = render(<Editor />);

  expect(container).toMatchSnapshot();
});
