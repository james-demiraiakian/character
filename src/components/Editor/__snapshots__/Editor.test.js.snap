// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of dom render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <div
          class="inputs"
        >
          <h1
            class="title"
          >
            Create A Character!
          </h1>
          <h5
            class="blurb"
          >
            You probably dont want to.
          </h5>
          <div
            class="form"
          >
            <label>
              Head: 
            </label>
            <select>
              <option
                value="placeholder"
              >
                 
              </option>
              <option
                value="head-1"
              >
                Head 1
              </option>
              <option
                value="head-2"
              >
                Head 2
              </option>
              <option
                value="head-3"
              >
                Head 3
              </option>
            </select>
          </div>
          <div
            class="form"
          >
            <label>
              Body: 
            </label>
            <select>
              <option
                value="placeholder"
              >
                 
              </option>
              <option
                value="body-1"
              >
                Body 1
              </option>
              <option
                value="body-2"
              >
                Body 2
              </option>
              <option
                value="body-3"
              >
                Body 3
              </option>
            </select>
          </div>
          <div
            class="form"
          >
            <label>
              Legs: 
            </label>
            <select>
              <option
                value="placeholder"
              >
                 
              </option>
              <option
                value="legs-1"
              >
                Legs 1
              </option>
              <option
                value="legs-2"
              >
                Legs 2
              </option>
              <option
                value="legs-3"
              >
                Legs 3
              </option>
            </select>
          </div>
          <div
            class="form"
          >
            <label>
              Phrase: 
            </label>
            <input
              type="text"
              value=""
            />
            <button>
              Save
            </button>
          </div>
        </div>
        <div
          class="counter"
        >
          <span
            class="head-count"
          >
            Your head has changed 
             times
          </span>
          <span
            class="body-count"
          >
            Your body has changed 
             times
          </span>
          <span
            class="legs-count"
          >
            Your legs have changed 
             times
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <div
        class="inputs"
      >
        <h1
          class="title"
        >
          Create A Character!
        </h1>
        <h5
          class="blurb"
        >
          You probably dont want to.
        </h5>
        <div
          class="form"
        >
          <label>
            Head: 
          </label>
          <select>
            <option
              value="placeholder"
            >
               
            </option>
            <option
              value="head-1"
            >
              Head 1
            </option>
            <option
              value="head-2"
            >
              Head 2
            </option>
            <option
              value="head-3"
            >
              Head 3
            </option>
          </select>
        </div>
        <div
          class="form"
        >
          <label>
            Body: 
          </label>
          <select>
            <option
              value="placeholder"
            >
               
            </option>
            <option
              value="body-1"
            >
              Body 1
            </option>
            <option
              value="body-2"
            >
              Body 2
            </option>
            <option
              value="body-3"
            >
              Body 3
            </option>
          </select>
        </div>
        <div
          class="form"
        >
          <label>
            Legs: 
          </label>
          <select>
            <option
              value="placeholder"
            >
               
            </option>
            <option
              value="legs-1"
            >
              Legs 1
            </option>
            <option
              value="legs-2"
            >
              Legs 2
            </option>
            <option
              value="legs-3"
            >
              Legs 3
            </option>
          </select>
        </div>
        <div
          class="form"
        >
          <label>
            Phrase: 
          </label>
          <input
            type="text"
            value=""
          />
          <button>
            Save
          </button>
        </div>
      </div>
      <div
        class="counter"
      >
        <span
          class="head-count"
        >
          Your head has changed 
           times
        </span>
        <span
          class="body-count"
        >
          Your body has changed 
           times
        </span>
        <span
          class="legs-count"
        >
          Your legs have changed 
           times
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
